sdk/python/feast/client.py

```diff
@@ -1181,85 +734,9 @@ def _get_grpc_metadata(self):
         return ()
 
 
-def _infer_online_entity_rows(
-    entity_rows: List[Dict[str, Any]],
-) -> List[GetOnlineFeaturesRequest.EntityRow]:
-    """
-    Builds a list of EntityRow protos from Python native type format passed by user.
-
-    Args:
-        entity_rows: A list of dictionaries where each key is an entity and each value is
-            feast.types.Value or Python native form.
-
-    Returns:
-        A list of EntityRow protos parsed from args.
-    """
-    entity_rows_dicts = cast(List[Dict[str, Any]], entity_rows)
-    entity_row_list = []
-    entity_type_map = dict()
-
-    for entity in entity_rows_dicts:
-        fields = {}
-        for key, value in entity.items():
-            # Allow for feast.types.Value
-            if isinstance(value, Value):
-                proto_value = value
-            else:
-                # Infer the specific type for this row
-                current_dtype = python_type_to_feast_value_type(name=key, value=value)
-
-                if key not in entity_type_map:
-                    entity_type_map[key] = current_dtype
-                else:
-                    if current_dtype != entity_type_map[key]:
-                        raise TypeError(
-                            f"Input entity {key} has mixed types, {current_dtype} and {entity_type_map[key]}. That is not allowed. "
-                        )
-                proto_value = _python_value_to_proto_value(current_dtype, value)
-            fields[key] = proto_value
-        entity_row_list.append(GetOnlineFeaturesRequest.EntityRow(fields=fields))
-    return entity_row_list
-
-
-def _build_feature_references(
-    feature_ref_strs: List[str], project: Optional[str] = None
-) -> List[FeatureReference]:
-    """
-    Builds a list of FeatureReference protos from string feature set references
-
-    Args:
-        feature_ref_strs: List of string feature references
-        project: Optionally specifies the project in the parsed feature references.
-
-    Returns:
-        A list of FeatureReference protos parsed from args.
-    """
-    feature_refs = [FeatureRef.from_str(ref_str) for ref_str in feature_ref_strs]
-    feature_ref_protos = [ref.to_proto() for ref in feature_refs]
-    # apply project if specified
-    if project is not None:
-        for feature_ref_proto in feature_ref_protos:
-            feature_ref_proto.project = project
-    return feature_ref_protos
-
-
-def _generate_ingestion_id(feature_set: FeatureSet) -> str:
-    """
-    Generates a UUID from the feature set name, version, and the current time.
-
-    Args:
-        feature_set: Feature set of the dataset to be ingested.
-
-    Returns:
-        UUID unique to current time and the feature set provided.
-    """
-    uuid_str = f"{feature_set.name}_{int(time.time())}"
-    return str(uuid.uuid3(uuid.NAMESPACE_DNS, uuid_str))
-
-
 def _read_table_from_source(
```

Comment:
Should write data partitioned by `date_partition_column` if it's specified.

Reply:
Updated.

Reply:
Sorry, I wasn't clear enough.
So current flow is:
1. Read from file / pandas
2. Write to parquet
3. Read from parquet
4. Write to parquet with partitions

Shouldn't it be only 1 and 4 when partitioning column specified?
Can all this flow be in one function?

Reply:
OK, I agree, we already have big functions that do a lot of stuff.
So what about? 
_read_table_from_source() -> returns Dataframe
if file:
     _write_to_parquet(df, timestamp_column, date_partitioning_column=None) -> returns directory
     recursively copy directory
if bq:
    load table from dataframe

Reply:
Current flow:

If FileSource and partition:
  1. Read from file/pandas
  2. Write to parquet with partitions
Else:
  1. Read from file / pandas
  2. Write to parquet